test/integration/test_components_error.py

```diff
@@ -68,3 +71,76 @@ def package_info(self):
     c.save({"conanfile.py": conanfile})
     c.run("create .", assert_error=True)
     assert 'The expected type for test_property is "list", but "str" was found' in c.out
+
+
+@pytest.mark.parametrize("component", [True, False])
+def test_unused_requirement(component):
+    """ Requires should include all listed requirements
+        This error is known when creating the package if the requirement is consumed.
+    """
+
+    t = TestClient(light=True)
+    conanfile = textwrap.dedent(f"""
+        from conan import ConanFile
+        class Consumer(ConanFile):
+            name = "wrong"
+            version = "version"
+            requires = "top/version", "top2/version"
+
+            def package_info(self):
+                self.cpp_info{'.components["foo"]' if component else ''}.requires = ["top::other"]
+    """)
+    t.save({"top/conanfile.py": GenConanfile().with_package_info({"components": {"cmp1": {"libs": ["top_cmp1"]}}}),
+            "conanfile.py": conanfile})
+    t.run('create top --name=top --version=version')
+    t.run('create top --name=top2 --version=version')
+    t.run('create .', assert_error=True)
+    assert "ERROR: wrong/version: Required package 'top2' not in component 'requires" in t.out
```

Comment:
This movement of the test makes impossible to see if the error message changed or what. Maybe better to leave the tests in their original place and use a separate, pure refactor PR to move these tests to the right location